Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/CachedQueries/Driver/GoHomeRequest.hs

```diff
@@ -0,0 +1,156 @@
+import Storage.CachedQueries.CacheConfig (CacheFlow)
+import qualified Storage.CachedQueries.GoHomeConfig as CQGHC
+import Storage.CachedQueries.Merchant.TransporterConfig as CQTC
+import Storage.Queries.Driver.GoHomeFeature.DriverGoHomeRequest as BeamDHR
+import Storage.Queries.Ride as Ride
+import Tools.Error (GenericError (..))
+
+makeGoHomeReqKey :: Id DP.Driver -> Text
+makeGoHomeReqKey = pack . ("CachedQueries:GoHomeRequest-driverId:" <>) . show
+
+getDriverGoHomeRequestInfo :: (CacheFlow m r, MonadFlow m, EsqDBFlow m r) => Id DP.Driver -> Id Merchant -> m CachedGoHomeRequest
+getDriverGoHomeRequestInfo driverId merchantId = do
+  initCnt <- CQGHC.findByMerchantId merchantId <&> (.startCnt)
+  expTime <- fromIntegral <$> asks (.cacheConfig.configsExpTime)
+  let ghkey = makeGoHomeReqKey driverId
+  currTime <- getLocalCurrentTime =<< ((CQTC.findByMerchantId merchantId >>= fromMaybeM (InternalError "Transporter config for timezone not found")) <&> (.timeDiffFromUtc))
+  Hedis.safeGet ghkey >>= \case
+    Just ghrData -> do
+      if fromMaybe False $ liftM2 (||) (ghrData.validTill <&> (> currTime)) (pure ghrData.isOnRide)
+        then return ghrData
+        else checkInvalidReqData ghrData currTime ghkey driverId merchantId
+    Nothing -> do
+      Hedis.setExp ghkey (templateGoHomeData Nothing initCnt Nothing Nothing False Nothing currTime) expTime
+      return $ templateGoHomeData Nothing initCnt Nothing Nothing False Nothing currTime
+
+checkInvalidReqData :: (CacheFlow m r, MonadFlow m, EsqDBFlow m r) => CachedGoHomeRequest -> UTCTime -> Text -> Id DP.Driver -> Id Merchant -> m CachedGoHomeRequest
+checkInvalidReqData ghrData currTime ghkey driverId merchantId = do
+  expTime <- fromIntegral <$> asks (.cacheConfig.configsExpTime)
+  initCnt <- CQGHC.findByMerchantId merchantId <&> (.startCnt)
+  if ghrData.expiryTime > currTime
+    then case ghrData.validTill of
+      Just _ -> do
+        ghrId <- fromMaybeM (InternalError "GoHomeRequestId not present but valid Till set") ghrData.driverGoHomeRequestId
+        succRide <- Ride.findSuccRideByGoHomeRequestId ghrId
+        if isJust succRide
+          then do
+            deactivateDriverGoHomeRequest merchantId driverId (Just SUCCESS)
+            return (templateGoHomeData Nothing (ghrData.cnt - 1) Nothing Nothing False (Just ghrData.expiryTime) currTime)
+          else do
+            logDebug "Driver Deactivating GoHomeRequest"
+            deactivateDriverGoHomeRequest merchantId driverId (Just FAILED)
+            return (templateGoHomeData Nothing ghrData.cnt Nothing Nothing False (Just ghrData.expiryTime) currTime)
+      Nothing -> do
+        return ghrData
+    else do
+      whenJust (ghrData.driverGoHomeRequestId) BeamDHR.finishWithFailure --Failing it as a default case as it's already a new day and old count does not matter.
+      Hedis.setExp ghkey (templateGoHomeData Nothing initCnt Nothing Nothing False Nothing currTime) expTime
```

Comment:
This data will not only give status, vTill but also will maintain the count as we discussed. So is important to always have to so that we as well as frontend are always aware how many go to driver has.